File: src/main/java/io/github/oleksivio/telegram/bot/api/annotations/filter/telegram/InvoiceFilter.java
package io.github.oleksivio.telegram.bot.api.annotations.filter.telegram;

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.IntegerFilter;
import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter;
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface InvoiceFilter {

    AnnotationState status() default AnnotationState.ON;

    String[] validator() default {};

    StringFilter title() default @StringFilter(status = AnnotationState.OFF);

    StringFilter description() default @StringFilter(status = AnnotationState.OFF);

    StringFilter startParameter() default @StringFilter(status = AnnotationState.OFF);

    StringFilter currency() default @StringFilter(status = AnnotationState.OFF);

    IntegerFilter totalAmount() default @IntegerFilter(status = AnnotationState.OFF);

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/telegram/InvoiceFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.telegram

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.IntegerFilter
import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class InvoiceFilter(val status: AnnotationState = AnnotationState.ON, val validator: Array<String> = [], val title: StringFilter = StringFilter(status = AnnotationState.OFF), val description: StringFilter = StringFilter(status = AnnotationState.OFF), val startParameter: StringFilter = StringFilter(status = AnnotationState.OFF), val currency: StringFilter = StringFilter(status = AnnotationState.OFF), val totalAmount: IntegerFilter = IntegerFilter(status = AnnotationState.OFF))
